src/effects/native/bessel8lvmixeqeffect.cpp

```diff
@@ -187,25 +184,66 @@ void Bessel8LVMixEQEffect::processChannel(const ChannelHandle& handle,
     //    pOutput[i + 1] = pState->m_pBandBuf[i];
     //}
 
-    if (fLow != pState->old_low ||
-            fMid != pState->old_mid ||
-            fHigh != pState->old_high) {
-        SampleUtil::copy3WithRampingGain(pOutput,
-                pState->m_pLowBuf, pState->old_low, fLow,
-                pState->m_pBandBuf, pState->old_mid, fMid,
-                pState->m_pHighBuf, pState->old_high, fHigh,
-                numSamples);
-    } else {
+    if (fLow == pState->old_low &&
+            fMid == pState->old_mid &&
+            fHigh == pState->old_high) {
         SampleUtil::copy3WithGain(pOutput,
                 pState->m_pLowBuf, fLow,
                 pState->m_pBandBuf, fMid,
                 pState->m_pHighBuf, fHigh,
                 numSamples);
-    }
+    } else {
+        int copySamples = 0; 
+        int rampingSamples = numSamples; 
+        if ((fLow && !pState->old_low) ||
+                (fMid && !pState->old_mid) ||
+                (fHigh && !pState->old_high)) {
+            // we have just switched at least one filter on
+            // Hold of ramping for the group delay
+            if (pState->m_rampHoldOff == kRampDone) {
+                // multiply the group delay * 2 to ensure that the filter is settled
+                // it is actually at a factor of 1,8 at default setting
+                pState->m_rampHoldOff = pState->m_groupDelay * 2;
+                // ensure that we have at least 128 samples (our smallest buffer)
+                // for ramping 
+                int rampingSamples = numSamples - (pState->m_rampHoldOff % numSamples);
+                if (rampingSamples < 128) {
+                    pState->m_rampHoldOff += rampingSamples; 
+                } 
+            }
+
+            if (pState->m_rampHoldOff >= static_cast<int>(numSamples)) {
+                // ramping is don in one of the folowing calls
+                copySamples = numSamples;
+                rampingSamples = 0;
+            } else {
+                // ramping in this call 
+                copySamples = pState->m_rampHoldOff;
+                rampingSamples = numSamples - pState->m_rampHoldOff; 
+            }
+            pState->m_rampHoldOff -= copySamples; 
+
+            SampleUtil::copy3WithGain(pOutput,
+                    pState->m_pLowBuf, pState->old_low,
+                    pState->m_pBandBuf, pState->old_mid,
+                    pState->m_pHighBuf, pState->old_high,
+                    copySamples);
+        }
 
-    pState->old_low = fLow;
-    pState->old_mid = fMid;
-    pState->old_high = fHigh;
+        if (rampingSamples) {
+            SampleUtil::copy3WithRampingGain(&pOutput[copySamples],
+                    &pState->m_pLowBuf[copySamples], pState->old_low, fLow,
```

Comment:
this block is exactly the same as the block in the other eq, is there any way to refactor the two eqs so this duplication is removed?